apps.go

```diff
@@ -217,27 +202,28 @@ func (s *scaler) Scale(ctx context.Context, opts ScaleOpts) (*Process, error) {
 		return nil, &ValidationError{Err: fmt.Errorf("no releases for %s", app.Name)}
 	}
 
-	f, err := s.store.Formation(ProcessesQuery{Release: release})
-	if err != nil {
-		return nil, err
-	}
-
-	p, ok := f[t]
-	if !ok {
+	p := release.Process(t)
+	if p == nil {
 		return nil, &ValidationError{Err: fmt.Errorf("no %s process type in release", t)}
 	}
 
-	if err := s.Scheduler.Scale(ctx, release.AppID, string(p.Type), uint(quantity)); err != nil {
-		return nil, err
-	}
-
 	// Update quantity for this process in the formation
 	p.Quantity = quantity
 	if c != nil {
 		p.Constraints = *c
 	}
 
-	return p, s.store.ProcessesUpdate(p)
+	if err := processesUpdate(db, p); err != nil {
+		return nil, err
+	}
+
+	// If there are no changes to the process size, we can do a quick scale
+	// up, otherwise, we will resubmit the release to the scheduler.
+	if c == nil {
+		return p, s.Scheduler.Scale(ctx, release.AppID, string(p.Type), uint(quantity))
+	} else {
+		return p, s.releaser.Release(ctx, release)
```

Comment:
This also fixes the issue where scaling with a new process size (e.g. `emp scale web=2:256:20m`) would require a new deployment to take effect.